### PR TITLE
Rename TARGET{,_SRC,_INC} to target{,_src,_inc} in CMakeLists.txt files.

### PR DESCRIPTION
**Pull request by bangerth**
We are having this conversation over in ASPECT land (see https://github.com/geodynamics/aspect/pull/3000) about making cmake variable names lowercase. I think that's what we've generally done, and reserved uppercase for cmake command names and context-dependent keywords. But we're not consistent.

This is a patch that does this for the `CMakeLists.txt` files in the `examples/` directory. I have no idea whether that will break anything (I suspect it will, in the auto-setup for tutorial programs), but thought I'd float it here for discussion.

### CODE REVIEW

File: examples/doxygen/CMakeLists.txt [hunk 1]
```diff
@@ -25,5 +25,5 @@ PROJECT(doxygen_examples)
 FOREACH(ccfile ${sources})
   GET_FILENAME_COMPONENT(file ${ccfile} NAME_WE)
   ADD_EXECUTABLE(${file} ${ccfile})
-  DEAL_II_SETUP_TARGET(${file})
+  DEAL_II_SETUP_target(${file})
```
Comment (tamiko) [inline examples/doxygen/CMakeLists.txt]: Would you mind fixing up the macro names to be all uppercase?

**Comment by tamiko** [comment]
@bangerth The convention for all the cmake stuff under `cmake/`, etc., is that uppercase is reserved for variables with a meaning in global scope, such as `DEAL_II_LINKER_FLAGS`, `BOOST_LIBRARIES`, etc., whereas `_lowercase` is reserved for local variables (i.e. names that are safe to reuse).

**Comment by bangerth** [comment]
I see. I had (wrongly) recalled that the underscore alone indicated local scope. 

@tjhei is not excited about the corresponding change in ASPECT. I don't particularly care one way or the other now that I understand the system. What should I do? Move forward, or just abandon the effort? Would we have to change the autopilot to accept both upper and lowercase variable names to stay backward compatible?

**Comment by tamiko** [comment]
@bangerth Such a change in the examples would be entirely self contained. The macro mechanism in cmake is based on substitution. What capitalization you choose in the user cmake file doesn't influence the macro definition :-)

**Comment by tamiko** [comment]
I think I should give a bit more context here for discussion:
 - CMake traditionally uses uppercase fo almost all variables, see for example [1], or have a look at what variables `find_package` exports, etc.
 - It is considered to be modern style to lowercase macro and function names, for example
    ```
    find_package(deal.II REQUIRED)
    deal_ii_setup_target(foobar)
    if(EXISTS ${CMAKE_SOURCE_DIR}/tests/CMakeLists.txt)
    ...
    endif
    ```
    However, I was not a huge fan of this style because it just looks strange that the key part of a line (keywords macros, etc) is deemphasized whereas variables and context sensitive words are emphasized. So I capitalized keywords as well. I know that this looks archaic...
 - We follow a pretty strict variable naming style in the CMake script in deal.II that I would like to retain, i.e. I would prefer no to introduce `variable` in addition to `_variable` and `VARIABLE`.
 - I actually wrote out a style guide for CMake [2]. We can of course change all or any portion of it if someone feels strongly about it :-)

[1] https://cmake.org/cmake/help/v3.0/manual/cmake-variables.7.html
[2] https://www.dealii.org/current/developers/cmake-internals.html#codingstyle

**Comment by bangerth** [comment]
Let's not worry about it then! I didn't realize the naming scheme (and will have to adjust to it next time I write something -- I think I authored a rather substantial share of the cmake trickery in doc/, and it probably doesn't follow your style guide :-( )

**Pull request closed without merging**